HarmonyOS/Day3 Assingment/KanishkaHalder/LoginSignUpApp/entry/src/main/java/com/applib/loginsignupapp/slice/SignUpSlice.java

```diff
@@ -0,0 +1,111 @@
+    private static final HiLogLabel LABEL_LOG = new HiLogLabel(HiLog.DEBUG, 0xD001100, "SIGNUP_LOG");
+    private DataAbilityHelper dbHelper;
+
+    private TextFieldValidated firstName;
+    private TextFieldValidated lastName;
+    private TextFieldValidated emailId;
+    private TextFieldValidated phoneNumber;
+    private TextFieldValidated password;
+    private Button submitBtn;
+    private RadioContainer radioContainer;
+    @Override
+    protected void onStart(Intent intent) {
+        super.onStart(intent);
+        super.setUIContent(ResourceTable.Layout_ability_signup);
+//        "permissions": [
+//          "com.applib.loginsignupapp.DataAbilityShellProvider.PROVIDER"
+//        ],
+        dbHelper = DataAbilityHelper.creator(this,URI);
+
+
+        firstName = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_firstName);
+        lastName = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_lastName);
+        emailId = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_email);
+        phoneNumber = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_phone);
+        password = (TextFieldValidated) findComponentById(ResourceTable.Id_textField_password);
+        submitBtn = (Button) findComponentById(ResourceTable.Id_button_signup_signup_slice);
+        radioContainer = (RadioContainer) findComponentById(ResourceTable.Id_radio_container);
+        submitBtn.setClickedListener(new Component.ClickedListener() {
+            @Override
+            public void onClick(Component component) {
+
+                boolean isAllValidated = true;
+                ValuesBucket value = new ValuesBucket();
+                if(firstName.getText()!=null) {
+                    value.putString("firstName", firstName.getText());
+                    isAllValidated = firstName.isValidated() && isAllValidated;
+                }
+                if(lastName.getText()!=null) {
+                    value.putString("lastName", lastName.getText());
+                    isAllValidated = lastName.isValidated() && isAllValidated;
+                }
+                if(emailId.getText()!=null) {
+                    value.putString("email", emailId.getText());
+                    isAllValidated = emailId.isValidated() && isAllValidated;
+                }
+                if(phoneNumber.getText()!=null) {
+                    value.putString("phone", phoneNumber.getText());
+                    isAllValidated = phoneNumber.isValidated() && isAllValidated;
+                }
+                if(password.getText()!=null) {
+                    value.putString("password", password.getText());
+                    isAllValidated = password.isValidated() && isAllValidated;
+                }
+                RadioButton selectedButton = (RadioButton) findComponentById(radioContainer.getMarkedButtonId());
```

Comment:
This casting is dangerous!! It will lead to ClassCastException - when no radiobuttons are selected in the radioContainer and crash the application.[App crash scenario : Submit the form without selecting any of the radiobuttons. The app will crash]